### DISCUSSION TITLE
Pybricks beta Blockprogramm

### DISCUSSION THREAD

**Discussion by msw-home** [pybricks/support#1375] [Q&A]
Hallo, ich bin neu beim programmieren in Pybricks eingestiegen.
Das ist ein Programmausschnitt aus einem bestehenden Projekt, geschrieben unter der Mindstorms Software.
Wie kann man das in der Pybricks Software erstellen in Blockprogramm? 
Gibt es eigentlich einen Onlinelehrgang für Pybricks Blockprogrammierung?
  
![image](https://github.com/pybricks/support/assets/12326241/61c5957a-b7bc-4035-b754-f9138fc1b3ef)

**Comment by laurensvalk** [comment]
Hi! Welcome to the forum. You can paste a screenshot in your post directly, just like in your email.

Or click this button below a new message:

![image](https://github.com/pybricks/support/assets/12326241/74a8dcc3-bc85-4646-8829-9f8d9b4a04bf)


That's much easier to see than a word document. Thank you!

**Comment by laurensvalk** [comment]
To mimic the behavior of when ("wenn"), you can add a program starting point and wait for that condition. For example, this will make play a sound whenever certain buttons are pressed:

![image](https://github.com/pybricks/support/assets/12326241/a826b91c-680f-4299-8a6d-8fc0e03a0940)

**Reply by laurensvalk** [reply]
Maybe we should make a single "when condition" block :)

**Reply by msw-home** [reply]
![pybricks 29 01 2024](https://github.com/pybricks/support/assets/155910298/c1b34ffb-5f33-47ef-8e90-40632e47b5c3)
Moin, eine Frage zu der Funktion von Blöcken.
Wo liegt der Unterschied zwischen parallel und multitask until?

**Reply by laurensvalk** [reply]
It can be used for the same purpose in some cases, but the `multitask` block gives you a few more possibilities.

For example, sometimes you just want to multitask only a few blocks and then continue normally. So you can do this:

This program will:
- run the motor forwards
- run the motor backwards and play sounds at the same time
- display "done" when both the movement and the sound are done.

![image](https://github.com/pybricks/support/assets/12326241/2a0a3924-fafa-4192-8333-2ab6c5dfcb63)
